lib/ui/painting.dart

```diff
@@ -1635,8 +1747,11 @@ class FrameInfo extends NativeFieldWrapperClass2 {
   Duration get duration => Duration(milliseconds: _durationMillis);
   int get _durationMillis native 'FrameInfo_durationMillis';
 
+  Image? _cachedImage;
+
   /// The [Image] object for this frame.
-  Image get image native 'FrameInfo_image';
+  Image get image => _cachedImage ??= Image._(_image);
+  _Image get _image native 'FrameInfo_image';
 }
```

Comment:
we should definitely update the docs for FrameInfo, FrameInfo.image, and any APIs that return FrameInfo to say that the Image _inside_ the returned object needs to be explicitly disposed and that if a handle to that image is passed it must first be cloned and that a handle to the FrameInfo itself must never be passed.

...which altogether sounds rather frightening. Should FrameInfo similarly be made cloneable to at least make it possible to create a new copy rather than making it a hot potato object? Having to dispose a property of a returned object is pretty weird as an API, I'm sure that would be a cause of leaks.

Why is FrameInfo a NativeFieldWrapperClass2? It looks like it could just be created with a reference to _image and _durationMillis and be a pure Dart object.

Reply:
The only problem I can think of with making FrameInfo not a NativeFieldWrapperClass2 is that `image` will no longer be lazy, so someone who is trying to "seek" to a particular frame will have to decode all the images between (whereas now they might not have to).

However, as implemented, `image` is not actually lazy, and making this a "regular" Dart object we could probably still keep it lazy by just pointing it to a private getter on the `Codec` that does the native work.

I think if we do make it lazy, we don't have to tell people to dispose it unless they access it.

I'm not a huge fan of cloning FrameInfo. You should either use the image and then dispose it, or you should pass it on without touching the image and tell the next person to dispose it.

Reply:
Hm. Making `image` lazy is probably something we should do in a separate patch, if at all. It'd shift decode workloads from when you call `getNextFrame` to whenyou first access the `image`, and we'd have to make the `image` getter async to make it really make sense.

Reply:
Added some more documentation on `FrameInfo`, `FrameInfo.image`, and `Codec.getNextFrame`.